Core/Graphics/Source/Graphics.cpp

```diff
@@ -44,25 +44,26 @@ namespace Babylon
         DisableRendering();
     }
 
-    void* Graphics::Impl::GetNativeWindow()
+    WindowType Graphics::Impl::GetNativeWindow()
     {
         std::scoped_lock lock{m_state.Mutex};
-        return m_state.Bgfx.InitState.platformData.nwh;
+        return (WindowType)m_state.Bgfx.InitState.platformData.nwh;
     }
 
-    void Graphics::Impl::SetNativeWindow(void* nativeWindowPtr, void* windowTypePtr)
+    void Graphics::Impl::SetNativeWindow(WindowType nativeWindowPtr, void* windowTypePtr)
```

Comment:
I think the way we really want this to work is for `SetNativeWindow` to take just one parameter, and only for certain specializations of the type of that parameter have `pd.ndt` get set. Like if the type of the `nativeWindowPtr` is known at compile time to be a `SwapChainPanel*`, then the type of `windowTypePtr` is also known to be 2. Therefore from a public API standpoint, `windowTypePtr` should not be part of the API, but rather an implementation detail. We couldn't do that before because we didn't know anything about the type of `nativeWindowPtr` at compile time, but if we are fixing that in this PR, then we should probably figure out how to get `windowTypePtr` out of the contract as well. @bghgary thoughts on the best way to do this?

Reply:
I put this comment on `Graphics::Impl` but it should have been on `Graphics` itself (the public API). The same concept applies though.

Reply:
Maybe what would make the most sense would be for `Graphics.h` to `#include GraphicsPlatform.h` and have non-templated members (such as `CreateGraphics` and `UpdateWindow`), but for the implementation to always defer to `GraphicsImpl`, which does have templated members, and then have template specializations per platform. Then we could have (for example) a platform agnostic `void*` version of `CreateGraphics`, and then each platform could provide a specialization for their window types (which would match the types in `GraphicsPlatform.h`) that could call the standard `void*` version and then optionally do additional stuff (like in the case of Windows, it could additionally set `pd.ndt`).